### PR TITLE
[Merged by Bors] - chore(FieldTheory/RatFunc): split the file into several ones

### PR DESCRIPTION
**Pull request by faenuccio**
Following the discussion in #12245 I am splitting `FieldTheory/RatFunc` into several files.

A new folder `RatFunc` is created in the parent `FieldTheory` together with the files
* `Basic` containing all basic definitions (lines 1-295)
* `FieldStructure` containing results about the field structure with numerators and denominators (lines 296-1381)
* `FunLike` containing properties about rational functions as functions (lines 1382-1558)
* `Degree` with properties about the degree (lines 1560-1657)

Lines 1658 to the end are moved to `RingTheory/LaurentSeries` .

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Great to see more file splits, thank you for doing this! Friendly reminder that you can remove the style exception for `RatFunc` when you're done. :-)

**Comment by faenuccio** [comment]
> Great to see more file splits, thank you for doing this! Friendly reminder that you can remove the style exception for `RatFunc` when you're done. :-)

Great to see this is useful! What is this style exception you are speaking about?

**Comment by grunweg** [comment]
TL;DR: remove this line: https://github.com/leanprover-community/mathlib4/blob/f1d422f45ad94f00f9b1ed78b9e28f0fb3c26ab8/scripts/style-exceptions.txt#L51
Longer version: there's now a style linter which errors on files longer than 1500 lines. Exceptions can be added manually to `scripts/style-exceptions.txt`; for now, it's easiest to remove entries when files are split. (There was a bot for mathlib3 which updated these automatically; porting that bot to mathlib4 is stuck in review.)

**Comment by faenuccio** [comment]
@grunweg @Vierkantor  Apart from a couple of merging issues that I am going to fix, my great question concerns the `#align` commands. They are all broken (partly because I moved the files and they point to an old structure that is no longer in force). I have commented all the problematic ones, but what is the right way here?

File: Mathlib/RingTheory/LaurentSeries.lean
Comment (ScottCarnahan) [inline Mathlib/RingTheory/LaurentSeries.lean:38]: Do we get a substantial benefit from adding an explicit universe variable here?

Comment (faenuccio) [reply Mathlib/RingTheory/LaurentSeries.lean:38]: I don't think we get, it seems more coherent with the choice made in all files concerning Rational Functions. But I am of course very happy to revert it if you suggest so.

Comment (ScottCarnahan) [reply Mathlib/RingTheory/LaurentSeries.lean:38]: There is no need to revert.  I just wanted to know if there was an interesting reason.

**Comment by ScottCarnahan** [comment]
Thanks!

maintainer merge

**Comment by ScottCarnahan** [comment]
Sorry, it looks like I don't have permission to do maintainer merge.

File: Mathlib/FieldTheory/RatFunc/Basic.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/Basic.lean:9]: You should keep this in all the split files (for now)

File: Mathlib/FieldTheory/RatFunc/Basic.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/Basic.lean:69]: Not in this file! Please check all declarations in this list

File: Mathlib/FieldTheory/RatFunc/Degree.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/Degree.lean:54]: Restore

File: Mathlib/FieldTheory/RatFunc/Degree.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/Degree.lean:60]: Restore

File: Mathlib/FieldTheory/RatFunc/FieldStructure.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/FieldStructure.lean:11]: Keep

File: Mathlib/FieldTheory/RatFunc/FieldStructure.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/FieldStructure.lean:22]: This one is still here

File: Mathlib/FieldTheory/RatFunc/FunLike.lean
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/RatFunc/FunLike.lean:40]: Restore all

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

**Comment by adomani** [comment]
I just briefly looked at why `move-decls` posted nothing and it seems that it thinks that there are no unaccounted for declarations!

I can investigate this further later on, but I think that it would be useful if it posted a "All declarations are accounted for!" message in situations like this.

**Review by faenuccio — COMMENTED** [review]
> Not in this file! Please check all declarations in this list 

Thanks! I have checked all doc's and have corrected them.

**Review by Vierkantor — APPROVED** [review]
Overall this is great work and very useful, thanks!

I would like to complain about the naming scheme somewhat.
 * `RatFunc.FieldStructure` seems to me still very basic: perhaps `RatFunc.Basic` can become `Defs` and this can become `Basic`?
 * `RatFunc.FunLike`: I don't see any mention of `FunLike` in that file. Perhaps something like `RatFunc.AsPolynomial`?

**Comment by faenuccio** [comment]
> Overall this is great work and very useful, thanks!
> 
> I would like to complain about the naming scheme somewhat.
> 
> * `RatFunc.FieldStructure` seems to me still very basic: perhaps `RatFunc.Basic` can become `Defs` and this can become `Basic`?
> * `RatFunc.FunLike`: I don't see any mention of `FunLike` in that file. Perhaps something like `RatFunc.AsPolynomial`?

Done!

**Comment by Vierkantor** [comment]
Thanks! I think this is good to go after solving the merge conflict.

bors d+

**Pull request closed without merging**
